package/com.unity.formats.usd/Runtime/Scripts/IO/Skel/SkeletonImporter.cs

```diff
@@ -274,13 +275,35 @@ public static class SkeletonImporter {
 
       // Unity 2019 supports many-bone rigs, older versions of Unity only support four bones.
 #if UNITY_2019
+      if (!isConstant && mesh.vertexCount == faceVertexIndices.Length)  // vertices were unrolled
+      {
+        int[] newIndices = new int[faceVertexIndices.Length * indicesElementSize];
+        float[] newWeights = new float[faceVertexIndices.Length * weightsElementSize];
+        for (int i = 0; i < faceVertexIndices.Length; i++)
+        {
+          for (int idxSize = 0; idxSize < indicesElementSize; idxSize++)
```

Comment:
You can merge both loops together as indicesElementSize always equals weightsElementSize